griptape/utils/decorators.py

```diff
@@ -15,6 +17,16 @@
 )
 
 
+def with_context(wrapped: Callable) -> Callable:
+    ctx = contextvars.copy_context()
```

Comment:
So this is not griptape-context specific then, is it? Should/can it be?

Reply:
Like what if client code is using contextvars for something else and doesn't want it propagated? (Like could be a pretty weird/difficult to debug side effect if it did happen... like maybe another library is doing something with contextvars and the customer doesn't even realize....)